crates/router/src/core/customers.rs

```diff
@@ -202,7 +203,10 @@ pub async fn delete_customer(
 
     let updated_customer = storage::CustomerUpdate::Update {
         name: Some(REDACTED.to_string()),
-        email: Some(REDACTED.to_string().into()),
+        email: match Email::from_str(REDACTED) {
+            Ok(email) => Some(email),
+            Err(_) => None,
+        },
```

Comment:
This would always fail and would cause unexpected behavior when we're trying to redact customer details, could you add a check in `Email::from_str()` that allows `REDACTED` as a valid email?

Reply:
Changes done. Also added a test for checking the REDACTED value.